File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int x = lower_bound(nums.begin(), nums.end(), target) - nums.begin();
        int y = upper_bound(nums.begin(), nums.end(), target) - nums.begin() - 1;

        // Check if target is in the range
        if (x == nums.size() || nums[x] != target) {
            return {-1, -1};
        }

        return {x, y};
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = new int[2];
        ans[0] = -1;
        ans[1] = -1;

        int l = 0, r = nums.length - 1;

        while(l <= r){
            int m = l + (r - l) / 2;

            if(nums[m] == target){
                ans[0] = m;
                r = m - 1;
            }
            else if(nums[m] > target){
                r = m  - 1;
            }
            else{
                l = m + 1;
            }
        }

        l = 0; r = nums.length - 1;
        
        while(l <= r){
            int m = l + (r - l) / 2;

            if(nums[m] == target){
                ans[1] = m;
                l = m + 1;
            }
            else if(nums[m] > target){
                r = m  - 1;
            }
            else{
                l = m + 1;
            }
        }

        return ans;
    }
}